### ISSUE TITLE
Symmetry for EXX

### ISSUE THREAD

**Issue by maki49** [deepmodeling/abacus-develop#2881]
## Background
Currently, EXX calculation does not support symmetry=0 and 1, due to wrong DM(R) (for all the functionals) and wrong Vs (for HF and PBE0). If symmetry is supported, the PBE part calculation can be reduced to a rate of nks_ibz/nks_tot.
## To Implement
- symmetry=0, DM(R) and Vs (#2868)
- symmetry=1, DM(R)

**Comment by hongriTianqi** [comment]
@maki49 Has the PR fully solved this issue?

**Comment by maki49** [comment]
> @maki49 Has the PR fully solved this issue?

Not yet.

**Comment by maki49** [comment]
Our current conclusion is that "symmetry=1" cannot be supported for EXX, because in LCAO $\ket{\phi_{\mu,k}}=\sum_R\phi_\mu(r-\tau_\mu-R)e^{ik\cdot R}$ is orthogonal with $\ket{\phi_{\mu,gk}}$ where g is a group operation, so I failed to use the relation $c_k=S^{-1}(gk,k)S(gk)c_{gk}$ ( where $S_{\mu\nu}(k_1,k_2)\equiv\braket{\phi_{k_1}|\phi_ {k_2}}$ ) derived from group theory to restore dm(k) from dm(gk): I implemented $S(gk,k)$ and found it a non-invertible zero-matrix ...
In other words, the dm(k) of every k-point is indispensable for constructing dm(R) for EXX.

I close this issue with the conclusion above. If someday I find any solution to this problem, I will reopen it.
